### PR TITLE
Framework: Add lib/make-json-schema-parser

### PR DESCRIPTION
**Pull request by sirreal**
`makeParser` is a useful tool to validate and transform data beyond the data layer. Extract the function and related code into its own directory. Updates usage across calypso. 

Bonus:
- Update some jsdocs for improved accuracy and layout
- Sort and alphabetize imports.

This PR should include _no functional changes_.

## Testing
- Smoke test
- Check e2e results

### CODE REVIEW

**Comment by sirreal** [comment]
May be a good opportunity to add a few tests for `makeJsonSchemaParser`. I'm happy to come back and add them.

File: client/state/data-layer/wpcom-http/test/utils.js [hunk 1]
```diff
@@ -9,10 +9,10 @@ import {
 	getProgress,
 	dispatchRequest,
 	dispatchRequestEx,
-	makeParser,
 	reducer,
 	trackRequests,
 } from '../utils.js';
+import makeJsonSchemaParser from 'lib/make-json-schema-parser';
```
Comment (sirreal) [inline client/state/data-layer/wpcom-http/test/utils.js]: Most of these tests should just stub the parser now

**Comment by sirreal** [comment]
I'd botched the imports, they're fixed now.

I've also added tests for the function. It would be good to remove the data-layer util tests reliance on `makeJsonSchemaParser`

**Comment by sirreal** [comment]
I've tried to removed data-layer test reliance on `makeJsonSchemaParser` in b154845

This is now ready for review.

File: client/state/data-layer/wpcom-http/test/utils.js [hunk 2]
```diff
@@ -9,10 +9,10 @@ import {
 	getProgress,
 	dispatchRequest,
 	dispatchRequestEx,
-	makeParser,
 	reducer,
 	trackRequests,
 } from '../utils.js';
+import { SchemaError } from 'lib/make-json-schema-parser';
```
Comment (jsnajdr) [inline client/state/data-layer/wpcom-http/test/utils.js]: Can we replace `SchemaError` with just `Error` and remove the dependency on `lib/make-json-schema-parser` completely?

File: client/state/data-layer/wpcom/sites/jitm/index.js [hunk 3]
```diff
@@ -8,14 +8,14 @@ import { noop, get } from 'lodash';
 /**
  * Internal dependencies
  */
+import makeJsonSchemaParser from 'lib/make-json-schema-parser';
+import schema from './schema.json';
+import { clearJITM, insertJITM } from 'state/jitm/actions';
 import { dispatchRequest } from 'state/data-layer/wpcom-http/utils';
+import { getSelectedSiteId } from 'state/ui/selectors';
```
Comment (jsnajdr) [inline client/state/data-layer/wpcom/sites/jitm/index.js]: Do you sort the imports with some algorithm or editor plugin? Just curious :wink:

Comment (sirreal) [reply client/state/data-layer/wpcom/sites/jitm/index.js]: vim `:sort i` (case-insensitive sort) usually with visual selection.

I manually sort multi-line imports after single-line 🤷‍♂️

**Comment by sirreal** [comment]
Thanks @jsnajdr! I've applied your suggestions and will merge after CI comes back green 👍

**Comment by dmsnell** [comment]
Thanks @sirreal - I think it was my mistake to pass the optional arguments into the `makeParser` function because in practice I've seen people manually merge JSON schema rather than pass related schemas in through the `ext` option.

That is, I think some next steps would be to consider making the signature `makeParser( schema, transformer )`

**Comment by sirreal** [comment]
@dmsnell Yeah, the ordering isn't too ergonomic. `schemaOptions` might be nice to have at some point, but it's completely unused right now.

Let's take discussion to #24314

**Pull request merged**